klaviyo-demo/index.ts

```diff
@@ -0,0 +1,144 @@
+import serverless from "serverless-http";
+import express from "express";
+
+const app = express();
+
+interface Profile {
+  type: 'profile';
+  id?: string;
+  attributes: {
+    last_name?: string;
+    first_name?: string;
+    properties: object;
+  };
+}
+
+const KlaviyoHeaders = {
+    'accept': 'application/json',
+    'content-type': 'application/json',
+    'revision': '2023-02-22',
+    'authorization': `Klaviyo-API-Key ${process.env.KLAVIYO_PRIVATE_API_KEY}`,
+};
+
+const getVerificationDetails = async (
+  verificationId: string
+) => {
+  const res = await fetch(
+    `https://services.sheerid.com/rest/v2/verification/${verificationId}/details`, {
+    'method': 'GET',
+    'headers': {
+      'Authorization': `Bearer ${process.env.SHEERID_ACCESS_TOKEN}`,
+      'Content-Type': 'application/json',
+      'User-Agent': 'SheerID Klaviyo Integration 0.1',
+    },
+  });
+  return res.json();
+}
+
+const findProfile = async (email: string) => {
+  const res = await fetch(`https://a.klaviyo.com/api/profiles/?filter=equals(email,%22${email}%22)`, {
```

Comment:
i'd think email should be URL encoded. Not sure it matters for this demo, though.